libs/vkd3d/vkd3d_private.h

```diff
@@ -647,6 +647,12 @@ static inline unsigned int d3d12_descriptor_heap_uav_set_index(bool is_buffer)
     return 3 + (is_buffer ? 0 : 1);
 }
 
+static inline uint32_t d3d12_desc_heap_offset(const struct d3d12_desc *dst)
+{
+    size_t heap_start = (size_t)dst->heap->descriptors;
+    return ((size_t)dst - heap_start) / sizeof(struct d3d12_desc);
```

Comment:
Is this sizeof aligned to POT? Integer divide isn't that cool. Alternatively, we can store the offset directly.

Reply:
> Is this sizeof aligned to POT?

No, but storing the offset (alongside the pointer to the heap, which we already have anyway) works too.